### PR TITLE
Add timezone mount to bootstrap

### PR DESCRIPTION
**Pull request by Eeems**
- [x] Update checksum
- [x] Test on device

### CODE REVIEW

**Comment by Eeems** [comment]
@matteodelabre looks like the label check hasn't properly reported it's status?

**Comment by matteodelabre** [comment]
Looks like you made your PR against `stable`, so it’s using the workflow files from that branch, which do not include the `pr-labels` workflow yet.

**Comment by matteodelabre** [comment]
Fixing this requires rebasing your branch on `testing`, so I’ll let you do that.

**Comment by Eeems** [comment]
Looks like I'll have to fix this up and recreate it. We should probably merge stable into testing every so often to make it easier to rebase when we accidentally do this.

**Comment by Eeems** [comment]
@matteodelabre Okay, fixed

File: package/toltec-bootstrap/entware-reenable
Comment (matteodelabre) [inline package/toltec-bootstrap/entware-reenable:3]: I suggest making the variable naming more consistent:

```suggestion
systemd_opt_mount_path=/lib/systemd/system/opt.mount
```

Comment (matteodelabre) [inline package/toltec-bootstrap/entware-reenable]: I suggest making the variable naming more consistent:

```suggestion
systemd_opt_mount_path=/lib/systemd/system/opt.mount
```

File: package/toltec-bootstrap/entware-reenable
Comment (matteodelabre) [inline package/toltec-bootstrap/entware-reenable]: Why is this needed?

File: package/toltec-bootstrap/entware-reenable
Comment (matteodelabre) [inline package/toltec-bootstrap/entware-reenable]: Is the extra level of indirection really needed here? If not, we could also remove the `After=opt.mount` dependency above.

```suggestion
What=/home/root/.entware/share/zoneinfo
```

File: scripts/bootstrap/bootstrap
Comment (matteodelabre) [inline scripts/bootstrap/bootstrap:378]: Let’s keep the verbosity level consistent.

```suggestion
        entware-install
        log "Mounting timezone info over /usr/share/timezone"
```

**Review by matteodelabre — CHANGES_REQUESTED** [review]
The duplication between the bootstrap script and the toltec-bootstrap package is something we will need to fix as soon as possible. But this is not the focus of this PR.

File: package/toltec-bootstrap/entware-reenable
Comment (Eeems) [reply package/toltec-bootstrap/entware-reenable]: In case we change the location. I copied from the entware-mount method.

File: package/toltec-bootstrap/entware-reenable
Comment (Eeems) [reply package/toltec-bootstrap/entware-reenable]: Fair enough

File: package/toltec-bootstrap/entware-reenable
Comment (matteodelabre) [reply package/toltec-bootstrap/entware-reenable]: Assuming we changed the location in the future, this would not remove the old file (as opposed to what is done in `entware-mount`). I propose we wait until we actually change the location to add this block, then we’ll know what the old name is and will be able to clean it.

Comment (Eeems) [reply package/toltec-bootstrap/entware-reenable]: Sounds good

**Comment by matteodelabre** [comment]
Thanks @Eeems! I wanted to test your changes but I ran into [this error](https://github.com/toltec-dev/toltec/pull/204#issuecomment-758865591) which seems to have been introduced before your PR.

**Comment by matteodelabre** [comment]
Turns out `systemd` refuses to enable the bind mount unless it’s named after the target path (so `usr-share-zoneinfo.mount`). I changed that and also made sure that the new unit is cleaned up in `error-cleanup()`.

**Review by matteodelabre — DISMISSED** [review]
Tested and confirmed to work on my rM. Waiting on @Eeems to review my changes before merging.

File: package/toltec-bootstrap/entware-reenable
Comment (Eeems) [inline package/toltec-bootstrap/entware-reenable]:
```suggestion
    systemctl enable --now "$(basename "$systemd_opt_mount_path")" 2> /dev/null
```

**Comment by matteodelabre** [comment]
Well spotted. Not sure how I missed that during testing, I must have used a version where I hadn’t made the change yet.

**Review by matteodelabre — DISMISSED** [review]
Confirmed to work on my rM1.

**Comment by Eeems** [comment]
@raisjn could you test on your rM2?

**Comment by Eeems** [comment]
Test scenarios:

- Test upgrading the toltec-boostrap package
- Run bootstrap on existing install
- Run entware-reenable on existing install where it's disabled
- Run entware-reenable on existing install where it's already enabled
- Run bootstrap on device without entware/toltec

**Comment by LinusCDE** [comment]
on rM 2 in order of testing:

 - Run bootstrap on device without entware/toltec » :heavy_check_mark: 
 - Run bootstrap on existing install » :heavy_check_mark: 
 - Run entware-reenable on existing install where it's disabled » :x: see logs below
 - Run entware-reenable on existing install where it's already enabled » :x: see same logs below

```shell
reMarkable: ~/ bash eetest 
Downloading https://bin.entware.net/armv7sf-k3.2/Packages.gz
Updated list of available packages in /opt/var/opkg-lists/entware
Downloading https://toltec-dev.org/testing/Packages.gz
Updated list of available packages in /opt/var/opkg-lists/toltec
eetest: command substitution: line 75: syntax error near unexpected token `|'
eetest: command substitution: line 75: `| uniq)'
```

 - Test upgrading the toltec-boostrap package » (:heavy_check_mark:) Worked but also triggered a re-installation of entware for me.

**Comment by LinusCDE** [comment]
We also may want to trigger the timezone check when updating the `toltec-bootstrap` package. Haven't checked whether this is already done but seemed not to for me when `opkg update/upgrade`ing from stable to testing now at least.

**Comment by matteodelabre** [comment]
> * Run entware-reenable on existing install where it's disabled » x see logs below
> * Run entware-reenable on existing install where it's already enabled » x see same logs below
> 
> ```console
> reMarkable: ~/ bash eetest 
> Downloading https://bin.entware.net/armv7sf-k3.2/Packages.gz
> Updated list of available packages in /opt/var/opkg-lists/entware
> Downloading https://toltec-dev.org/testing/Packages.gz
> Updated list of available packages in /opt/var/opkg-lists/toltec
> eetest: command substitution: line 75: syntax error near unexpected token `|'
> eetest: command substitution: line 75: `| uniq)'
> ```

Interesting. I re-ran entware-reenable on my rM1 in both scenarios and I don’t get any error. Can you confirm that we have the same `entware-reenable` script? Mine has a SHA-256 sum of `701173003cda882067d97d2f7f9fb8a18d2c17ac888e9646dee5acda3f34ae67`. Can you also provide us with the Bash version that’s running on the rM2? On rM1, I have 4.4.12(1).

> * Test upgrading the toltec-boostrap package » (heavy_check_mark) Worked but also triggered a re-installation of entware for me.

I don’t think just upgrading toltec-bootstrap should trigger the reenable script, since the package itself doesn’t contain any install script. Can you confirm @Eeems?

**Comment by Eeems** [comment]
> I don’t think just upgrading toltec-bootstrap should trigger the reenable script, since the package itself doesn’t contain any install script. Can you confirm @Eeems?

Correct, this shouldn't happen.



> on rM 2 in order of testing:
> 
>     * Run bootstrap on device without entware/toltec » heavy_check_mark
> 
>     * Run bootstrap on existing install » heavy_check_mark
> 
>     * Run entware-reenable on existing install where it's disabled » x see logs below
> 
>     * Run entware-reenable on existing install where it's already enabled » x see same logs below
> 
> 
> ```shell
> reMarkable: ~/ bash eetest 
> Downloading https://bin.entware.net/armv7sf-k3.2/Packages.gz
> Updated list of available packages in /opt/var/opkg-lists/entware
> Downloading https://toltec-dev.org/testing/Packages.gz
> Updated list of available packages in /opt/var/opkg-lists/toltec
> eetest: command substitution: line 75: syntax error near unexpected token `|'
> eetest: command substitution: line 75: `| uniq)'
> ```
> 
>     * Test upgrading the toltec-boostrap package » (heavy_check_mark) Worked but also triggered a re-installation of entware for me.

Line 75 is `echo -e "${packages[@]}\n${available_packages[@]}" | sed 's/ /\0/g' | sort | uniq -d | awk '{print $1}' | xargs -r opkg install --force-reinstall`, which is a line I didn't touch. It also looks good to me. Perhaps `$available_packages` and `$packages` are empty?

**Comment by LinusCDE** [comment]
> Interesting. I re-ran entware-reenable on my rM1 in both scenarios and I don’t get any error. Can you confirm that we have the same `entware-reenable` script? Mine has a SHA-256 sum of `701173003cda882067d97d2f7f9fb8a18d2c17ac888e9646dee5acda3f34ae67`. Can you also provide us with the Bash version that’s running on the rM2? On rM1, I have 4.4.12(1).
> 

> Line 75 is `echo -e "${packages[@]}\n${available_packages[@]}" | sed 's/ /\0/g' | sort | uniq -d | awk '{print $1}' | xargs -r opkg install --force-reinstall`, which is a line I didn't touch. It also looks good to me. Perhaps `$available_packages` and `$packages` are empty?

Got the problem. My fault. I took the latest file from here, but copy&pasted the content into nano which made a long line wrap. If I fix that the file matches with your hash, Mattéo.

That was my change. Makes sense that the command didn't work:
```diff
72,73c72
<     available_packages=$(gunzip -c /opt/var/opkg-lists/* | grep -E '(Package:)|(Version:)' | sed 's|Package: ||' | tr '\n' ' ' | sed -r 's| Version: ([^ ]+) | \1\n|g' | sort 
< | uniq)
---
>     available_packages=$(gunzip -c /opt/var/opkg-lists/* | grep -E '(Package:)|(Version:)' | sed 's|Package: ||' | tr '\n' ' ' | sed -r 's| Version: ([^ ]+) | \1\n|g' | sort | uniq)
```

I re-tested:
 - Run entware-reenable on existing install where it's disabled
 - Run entware-reenable on existing install where it's already enabled

For my test, I had remux installed, disabled opt.mount and usr-share-zoneinfo.mount and removed these unit files as well. The first reenable didn't reinstall remux to add the file again. I added rm2fb and installed remux (which was listed as removed?). After that I ran the second test and it reinstalled remux and rm2fb.

I would say this is not an issue with this specific PR, but entware-reenable might not reinstall if the firmware gets updated and disables toltec/entware.

**Comment by Eeems** [comment]
> I would say this is not an issue with this specific PR, but entware-reenable might not reinstall if the firmware gets updated and disables toltec/entware.

Should we try to fix it as part of this, or as part of a different PR?

**Comment by LinusCDE** [comment]
> > I would say this is not an issue with this specific PR, but entware-reenable might not reinstall if the firmware gets updated and disables toltec/entware.
> 
> Should we try to fix it as part of this, or as part of a different PR?

Ok, scratch that as well. Redid the test with faking a fw update and now it worked fine. Not sure what I missed in the last test.

**Review by LinusCDE — APPROVED** [review]

**Pull request merged**
